pkgs/by-name/sm/smtp4dev/package.nix

```
@@ -27,11 +30,20 @@ buildDotnetModule {
     nodejs.python
     npmHooks.npmConfigHook
     stdenv.cc # c compiler is needed for compiling npm-deps
+    gnused
+    gnupatch
   ];
 
   npmDeps = fetchNpmDeps {
     src = "${src}/${npmRoot}";
-    hash = "sha256-/Z6sBxA2ReHlEbz0zJjlpn6IwzHDQiXN5ixEV1/iCJI=";
+    hash = "sha256-Uj0EnnsA+QHq5KHF2B93OG8rwxYrV6sEgMTbd43ttCA=";
+    postPatch = ''
+      # remove Rnwood.Smtp4dev/ClientApp/ so we can use this patch
+      cp ${./smtp4dev-npm-packages.patch} smtp4dev-npm-packages.patch
+      sed -i 's|Rnwood.Smtp4dev/ClientApp/||g' smtp4dev-npm-packages.patch
+      patch -p1 < smtp4dev-npm-packages.patch
+      rm smtp4dev-npm-packages.patch
+    '';
```

maybe do something like this instead, so you don't have to patch the patch

```suggestion
    inherit src patches;
    postPatch = "cd ${npmRoot}";
```

---

The patch is applied inside the client folder of the subdirectory, so we need to remove the path from the patch or it won't apply unfortunately.

---

My idea was to pass `src` directly to `fetchNpmDeps` instead of only the client folder, so the patch can be applied the same way it is in the main derivation.

---

Seems like this has already been done here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/rstudio/default.nix#L187-L193